Ticketing/Ticketing.BAL/Services/CartService.cs

```diff
@@ -0,0 +1,116 @@
+
+        public CartService(Repository<ShoppingCart> repository, Repository<Seat> repositorySeat, Repository<Payment> repositoryPayment)
+        {
+            _repositoryShoppingCart = repository;
+            _repositorySeat = repositorySeat;
+            _repositoryPayment = repositoryPayment;
+        }
+
+        public async Task<CartStateReturnModel> AddSeatToCartAsync(Guid CartId, OrderCartModel orderCartModel)
+        {
+
+            var cartId = CartId;
+
+            var shoppingCarts = await _repositoryShoppingCart.GetAllAsync();
+
+            var item = shoppingCarts.Where(c => c.CartId == cartId && c.EventId == orderCartModel.EventId && c.SeatId == orderCartModel.SeatId).FirstOrDefault();
+
+            var shoppingCartDto = new ShoppingCart
+            {
+                EventId = orderCartModel.EventId,
+                SeatId = orderCartModel.SeatId,
+                PriceTypeId = orderCartModel.PriceTypeId,
+                Price = orderCartModel.Price,
+                CartId = CartId,
+            };
+
+            if (item is null)
+            {
+                await _repositoryShoppingCart.CreateAsync(shoppingCartDto);
+            }
+            else
+            {
+                item.PriceTypeId = orderCartModel.PriceTypeId;
+                item.Price = orderCartModel.Price;
+            }
+
+            await _repositoryShoppingCart.UpdateAsync(shoppingCartDto);
+
+            shoppingCarts = await _repositoryShoppingCart.GetAllAsync();
+
+            var totalAmount = shoppingCarts.Sum(sc => sc.Price);
+
+            return new CartStateReturnModel
+            {
+                CartId = cartId,
+                TotalAmount = totalAmount
+            };
+        }
+
+        public async Task<int> BookSeatToCartAsync(Guid cartId)
+        {
+            var shoppingCarts = await _repositoryShoppingCart.GetAllAsync();
+
+            var shoppingCartItems = shoppingCarts.Where(c => c.CartId == cartId).ToList();
+
+            var shoppingCartSeats = shoppingCartItems.Select(sh => sh.SeatId).ToList();
+
+            var allSeats = await _repositorySeat.GetAllAsync();
+
+            var seats = allSeats.Where(s => shoppingCartSeats.Contains(s.Id));
+
+            decimal totalAmount = shoppingCartItems.Sum(i => i.Price);
+
+            foreach (var seat in seats)
+            {
+                seat.SeatStatusId = SeatState.Booked;
+                await _repositorySeat.UpdateAsync(seat);
+            }
+
+            var payment = new Payment
+            {
+                Amount = totalAmount,
+                CartId = cartId,
+                PaymentStatusId = PaymentState.NoPayment
+            };
+
+            var newPayment = await _repositoryPayment.CreateAsync(payment);
+
+            return newPayment.Id;
+        }
+
+        public async Task DeleteSeatForCartAsync(Guid cartId, int eventId, int seatId)
+        {
+            var shoppingCarts = await _repositoryShoppingCart.GetAllAsync();
```

Comment:
to the one LINQ please. Also ToListAsync for all places like that

Reply:
it doesn't work that way